### PR TITLE
Genesis function and genesis block validation

### PR DESCRIPTION
**Pull request by david-julien**

### CODE REVIEW

File: blockchain/test_utils.go [hunk 1]
```diff
@@ -234,22 +234,3 @@ func NewValidCloudBaseTransaction() (*Transaction, Address) {
 	}
 	return cbTx, w.Public()
 }
-
-// BigIntToHash converts a big integer to a hash
-func BigIntToHash(x *big.Int) Hash {
```
Comment (david-julien) [inline blockchain/test_utils.go:239]: Realized I was using this function to do some math in consensus.go, and thus moved it to hash.go.

File: blockchain/genesis.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package blockchain
+
+import (
+	"time"
+)
+
+// Genesis creates the Genesis block and returns is.
```
Comment (david-julien) [inline blockchain/genesis.go]: Take a look at the genesis block properties. These are pretty standard across cryptocurrencies. Let me know if you think anything needs to be added.

File: blockchain/genesis.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+//
+// Properties of the Genesis block:
```
Comment (david-julien) [inline blockchain/genesis.go:9]: Take a look at the genesis block properties. These are pretty standard across cryptocurrencies. Let me know if you think anything needs to be added.

File: blockchain/hash.go [hunk 4]
```diff
@@ -57,3 +57,22 @@ func HashToBigInt(h Hash) *big.Int {
 	}
 	return new(big.Int).SetBytes(h[:])
 }
+
+// BigIntToHash converts a big integer to a hash
+func BigIntToHash(x *big.Int) Hash {
```
Comment (jordanschalm) [inline blockchain/hash.go:62]: Maybe call this `FromBigInt` so you call it like `hash.FromBigInt`

File: blockchain/validation.go [hunk 5]
```diff
@@ -26,7 +31,7 @@ const (
 	BadSig TransactionCode = iota
```
Comment (jordanschalm) [inline blockchain/validation.go]: Just need to write `iota` once in each `const` block

File: blockchain/validation.go [hunk 6]
```diff
@@ -52,14 +57,37 @@ const (
 )
 
 const (
+	// ValidGenesisBlock is returned when a block is a valid genesis block.
+	ValidGenesisBlock GenesisBlockCode = iota
```
Comment (jordanschalm) [inline blockchain/validation.go]: same, just write iota once (for the first declaration)

File: blockchain/validation.go [hunk 7]
```diff
@@ -52,14 +57,37 @@ const (
...
+	BadGenesisTarget GenesisBlockCode = iota
+	// BadGenesisTime is returned when teh gensis block's time is invalid.
+	BadGenesisTime GenesisBlockCode = iota
+	// NilGenesisBlock is returned when the genesis block is equal to nil.
+	NilGenesisBlock GenesisBlockCode = iota
+)
+
+const (
 	// ValidBlock is returned when the block is valid.
 	ValidBlock BlockCode = iota
```
Comment (jordanschalm) [inline blockchain/validation.go]: write iota once

**Review by jordanschalm — APPROVED** [review]
🚢

**Pull request merged**